include/graphene-box2d.h

```diff
@@ -0,0 +1,122 @@
+/* graphene-box2d.h: An 2D, axis aligned bounding box
+ *
+ * SPDX-License-Identifier: MIT
+ * SPDX-FileCopyrightText: 2023  Emmanuele Bassi
+ */
+
+#pragma once
+
+#if !defined(GRAPHENE_H_INSIDE) && !defined(GRAPHENE_COMPILATION)
+#error "Only graphene.h can be included directly."
+#endif
+
+#include "graphene-types.h"
+#include "graphene-vec2.h"
+
+GRAPHENE_BEGIN_DECLS
+
+/**
+ * graphene_box2d_t:
+ *
+ * A 2D box, described as the axis-aligned area between a minimum and
+ * a maximum vertices lying on the same plane.
+ *
+ * Since: 1.12
+ */
+struct _graphene_box2d_t
+{
+  /*< private >*/
+  GRAPHENE_PRIVATE_FIELD (graphene_vec2_t, min);
+  GRAPHENE_PRIVATE_FIELD (graphene_vec2_t, max);
+};
+
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_alloc                      (void);
+GRAPHENE_AVAILABLE_IN_1_12
+void                      graphene_box2d_free                       (graphene_box2d_t       *box);
+
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init                       (graphene_box2d_t       *box,
+                                                                     const graphene_point_t *min,
+                                                                     const graphene_point_t *max);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_points           (graphene_box2d_t       *box,
+                                                                     unsigned int            n_points,
+                                                                     const graphene_point_t *points);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_vectors          (graphene_box2d_t       *box,
+                                                                     unsigned int            n_vectors,
+                                                                     const graphene_vec2_t  *vectors);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_box              (graphene_box2d_t       *box,
+                                                                     const graphene_box2d_t *src);
+GRAPHENE_AVAILABLE_IN_1_12
+graphene_box2d_t *        graphene_box2d_init_from_vec2             (graphene_box2d_t       *box,
+                                                                     const graphene_vec2_t  *min,
+                                                                     const graphene_vec2_t  *max);
```

Comment:
Transitioning GTK code from using `graphene_rect_t` would benefit from `graphene_box2d_init_from_rect()` and `graphene_rect_init_from_box2d()`.